src/test/runtime/test_node_path.py

```diff
@@ -0,0 +1,169 @@
+                ]
+            )
+        )
+        records = node_path.to_path_beginning_records()
+        expected = Records(
+            [
+                Record({
+                    'callback_start/0': 0, 'pub/0': 1
+                }),
+            ],
+            [
+                ColumnValue('callback_start/0'),
+                ColumnValue('pub/0')
+            ]
+        )
+
+        assert records.equals(expected)
+
+    def test_end_path(self, mocker):
+        records_provider = mocker.Mock(spec=RecordsProvider)
+        node_path_value = mocker.Mock(spec=NodePathStructValue)
+        node_path = NodePath(
+            node_path_value,
+            records_provider,
+            subscription=None,
+            publisher=None,
+            callbacks=None
+        )
+
+        mocker.patch.object(
+            records_provider, 'path_end_records',
+            return_value=Records(
+                [
+                    Record({
+                        'callback_start/0': 0, 'callback_end/0': 1
+                    }),
+                ],
+                [
+                    ColumnValue('callback_start/0'),
+                    ColumnValue('callback_end/0')
+                ]
+            )
+        )
+        records = node_path.to_path_end_records()
+
+        expected = Records(
+            [
+                Record({
+                    'callback_start/0': 0, 'callback_end/0': 1
+                }),
+            ],
+            [
+                ColumnValue('callback_start/0'),
+                ColumnValue('callback_end/0')
+            ]
+        )
+        assert records.equals(expected)
+
+    def test_to_records(self, mocker):
+        records_provider = mocker.Mock(spec=RecordsProvider)
+        node_path_value = mocker.Mock(spec=NodePathStructValue)
+        node_path = NodePath(
+            node_path_value,
+            records_provider,
+            subscription=None,
+            publisher=None,
+            callbacks=None
+        )
+
+        mocker.patch.object(
+            records_provider, 'node_records',
+            return_value=Records(
+                [
+                    Record({
+                        'callback_start': 0, 'xxx': 1, 'pub': 2
+                    }),
+                ],
+                [
+                    ColumnValue('callback_start'),
+                    ColumnValue('xxx'),
+                    ColumnValue('pub'),
+                ]
+            )
+        )
+        records = node_path.to_records()
+
+        expected = Records(
+                [
+                    Record({
+                        'callback_start': 0, 'xxx': 1, 'pub': 2
+                    }),
+                ],
+                [
+                    ColumnValue('callback_start'),
+                    ColumnValue('xxx'),
+                    ColumnValue('pub'),
+                ]
+            )
+
+        assert records.equals(expected)
+
+    def test_cache(self, mocker):
```

Comment:
@ymski It is better to test caching of to_path_beginning_records() and to_path_end_records().